### ISSUE TITLE
Running org.scalatest.tools.Runner with an invalid Reporter shows exception but then hangs

### ISSUE THREAD

**Issue by jackkoenig** [scalatest/scalatest#2069]
I have a custom reporter that I provide to scalafmt using the CLI (no SBT or other build tool involved). This is a bit problematic because if I screw something up in CI, my CI just hangs. It would be ideal if the `Runner` would exit with a non-zero code rather than hanging.

**Comment by cheeseng** [comment]
@jackkoenig Sorry for replying late on this, I am curious what is causing the hang, is it still possible to get the stacktrace of the last error when it hangs?

Cheers.

**Comment by jackkoenig** [comment]
No worries and thanks for your response @cheeseng! I should have included it before, here's the stack trace:
```
An exception or error caused a run to abort. This may have been caused by a problematic custom reporter.
java.lang.IllegalArgumentException: Couldn't load a Reporter class: "my.custom.ReporterButMisspelt".
Problem arg: -r... my.custom.ReporterButMisspelt
        at org.scalatest.tools.ReporterFactory.getCustomReporter(ReporterFactory.scala:39)
        at org.scalatest.tools.ReporterFactory.createCustomReporter(ReporterFactory.scala:213)
        at org.scalatest.tools.ReporterFactory.getReporterFromConfiguration(ReporterFactory.scala:246)
        at org.scalatest.tools.ReporterFactory.$anonfun$createReportersFromConfigurations$1(ReporterFactory.scala:254)
        at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
        at scala.collection.Iterator.foreach(Iterator.scala:943)
        at scala.collection.Iterator.foreach$(Iterator.scala:943)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
        at scala.collection.IterableLike.foreach(IterableLike.scala:74)
        at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
        at org.scalatest.tools.ReporterConfigurations.foreach(ReporterConfiguration.scala:42)
        at scala.collection.TraversableLike.map(TraversableLike.scala:286)
        at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
        at org.scalatest.tools.ReporterConfigurations.map(ReporterConfiguration.scala:42)
        at org.scalatest.tools.ReporterFactory.createReportersFromConfigurations(ReporterFactory.scala:253)
        at org.scalatest.tools.ReporterFactory.getDispatchReporter(ReporterFactory.scala:257)
        at org.scalatest.tools.Runner$.withClassLoaderAndDispatchReporter(Runner.scala:1480)
        at org.scalatest.tools.Runner$.runOptionallyWithPassFailReporter(Runner.scala:971)
        at org.scalatest.tools.Runner$.main(Runner.scala:775)
        at org.scalatest.tools.Runner.main(Runner.scala)
Caused by: java.lang.ClassNotFoundException: <my.custom.ReporterButMisspelt>
        at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        at org.scalatest.tools.ReporterFactory.getCustomReporter(ReporterFactory.scala:29)
        ... 19 more
```

This is using scalatest 3.2.6 but the same issue manifests with 3.2.10.

**Comment by cheeseng** [comment]
@jackkoenig Fyi I submitted the following PR for this: 

https://github.com/scalatest/scalatest/pull/2090

**Comment by jackkoenig** [comment]
Thank you!

**Comment by cheeseng** [comment]
@jackkoenig Fyi this should be fixed now in 3.2.11, do you mind to verify and close this ticket if it works?

**Comment by jackkoenig** [comment]
I can confirm this is indeed fixed. Thank you!
